### ISSUE TITLE
STATION/TRAJECTORY _PARAMETER rules review

### ISSUE THREAD

**Issue by MarkIggy** [euroargodev/ArgoNetCDF#21]
**Review of the FileChecker rules for STATION_PARAMETERS and TRAJECTORY_PARAMETERS.**

Referencing the "Description of Argo GDAC File Checks"

***PROPOSED CHANGES:***  (in bold)

Section 5.3 -- Profile files
STATION_PARAMETERS:
S1: All valid parameter names. Reference table 3.
S2: For each parameter, <PARAM> variable exists in file.
S3: For each <PARAM> variable **~~(with data)~~**, parameter name included in STATION_PARAMETERS
S4: Core-file / profile #1: PRES and TEMP present
S5: No duplicate names in list.
S6: No blank entries within the sequence of names. (For example, “PRES”, “   “, “TEMP”).    **Reject**

Section 7.1 -- Trajectory files
TRAJECTORY_PARAMETERS	
T1: All valid parameter names. Reference table 3.
T2: For each parameter, <PARAM> variable exists in file.
T3: For each <PARAM> variables **~~(with data)~~**, parameter name included in TRAJECTORY_PARAMETERS
T4: No duplicate names in list.
**~~T5: Blank entries within the sequence of names. (For example, “PRES”, “   “, “TEMP”).~~**
**T5: No blank entries in the list.**

**DISCUSSION:**

Rules S3 and T3:  The original rules allowed for the situation where the name of a  <PARAM> variable in the file did not need to be in the *_PARAMETER list if all of its values were _FillValue.  With this revision: If a <PARAM> variable is in a file, it **must** be listed in the *_PARAMETERS variable.

Rule S6: Blank entries must be allowed in profile files because of the way multiple profiles are recorded within the file.  Rule S6 says the blank entries must be at the end of the list.  Previously, this was only a warning.  Propose changing it to a rejection.

Rule T5:  There is no reason to have blank entries in TRAJECTORY_PARAMETERS.  So disallow them.

**Comment by jegilson** [comment]
I agree with implementation of these changes.

**Comment by ClaudiaAOML** [comment]
I do not agree with S6 resulting in a rejection. That is a cosmetic change (we have one particular float type that has this issue for BR-files, because of the complicated way it reports some data only in NPROF=1 and others only in NPROF=2). It may seem like a simple change (I wish ...). Problem is: we have a lot of more important updates in the pipeline and more to come because of all the BGC updates that were decided at the previous ADMY meeting.

**Comment by ClaudiaAOML** [comment]
I wanted to add (and hit the wrong keys to insert a line feed): withholding good profile data because of something that looks weird but is not wrong does not seem like a good idea.

**Comment by ClaudiaAOML** [comment]
If T5 does not currently result in a warning then I recommend to start with a warning.

If a warning is issued now then a transition to reject seems OK, because it seems like a much easier thing to fix than in the profile files.

**Comment by mscanderbeg** [comment]
If S6 is too hard to accommodate currently, let's keep it as a warning for now and re-evaluate at the ADMT-24 meeting.  
Otherwise, I agree with the suggested changes.
